src/awkward/_v2/_connect/avro.py

```diff
@@ -0,0 +1,1440 @@
+            _exec_code.append(
+                "\n"
+                + "    " * ind
+                + f"con['node{count}-offsets'].append(out+con['node{count}-offsets'][-1])"
+            )
+            _exec_code.append("\n" + "    " * ind + "if out < 0:")
+            _exec_code.append(
+                "\n" + "    " * (ind + 1) +
+                "pos, inn = decode_varint(pos,fields)"
+            )
+            _exec_code.append("\n" + "    " * (ind + 1) +
+                              "nbytes = decode_zigzag(inn)")
+            _exec_code.append("\n" + "    " * ind + "for j in range(out):")
+            aform, _exec_code, count, dec = self.rec_exp_json_code(
+                {"type": file["items"]}, _exec_code, ind +
+                1, aform, count + 1, dec
+            )
+            _exec_code.append(
+                "\n" + "    " * ind + "pos, inn = decode_varint(pos,fields)"
+            )
+            _exec_code.append("\n" + "    " * ind + "out = decode_zigzag(inn)")
+            _exec_code.append("\n" + "    " * ind + """if out != 0:""")
+            _exec_code.append("\n" + "    " * (ind + 1) + "raise")
+            aform.append(f'"form_key": "node{temp}"}},\n')
+            return aform, _exec_code, count, dec
+
+        elif file["type"] == "map":
+            # print(file["name"])
+            #         _exec_code.append("\npos, inn = decode_varint(pos,fields)"
+            #         _exec_code.append("\nout = abs(decode_zigzag(inn))"
+            #         _exec_code.append("\nprint(\"length\",out)"
+            #         _exec_code.append("\nfor i in range(out):"
+            #         _exec_code.append("\n"+"    "*(ind+1)+"print(\"{{\")"
+            #         _exec_code = _exec_code+aa
+            #         _exec_code = _exec_code+bb
+            #         _exec_code = _exec_code+ccfa
+            #         _exec_code = _exec_code+dd
+            #         _exec_code = _exec_code+ee
+            #         pos,_exec_code,count = self.rec_exp_json_code({"type": file["values"]},fields,pos,_exec_code,ind+1, aform,count+1)
+            #         _exec_code.append("\n    "*(ind+1)+"print(\":\")"
+            #         _exec_code = _exec_code+ff
+            #         pos,_exec_code,count = self.rec_exp_json_code({"type": file["values"]},fields,pos,_exec_code,ind+1, aform,count+1)
+            #         _exec_code.append("\n"+"    "*(ind+1)+"print(\"}}\")"
+            #         _exec_code.append("\n"+"    "*ind+"pos, inn = decode_varint(pos,fields)"
+            #         jj = "\n"+"    "*ind+"out = decode_zigzag(inn)"
+            #         kk = "\n"+"    "*ind+'''if out != 0:
+            #     raise
+            #             '''
+            #         _exec_code = _exec_code+gg
+            #         _exec_code = _exec_code+hh
+            #         _exec_code = _exec_code+jj
+            #         _exec_code = _exec_code+kk
+            raise NotImplementedError
+
+
+class LengthError(Exception):
+    pass
```

Comment:
Internal: should start with underscore.

Should be more descriptive. More like `ReachedEndOfArray`?